### DISCUSSION TITLE
Naming convention for API paths, query parameters, variables etc.

### DISCUSSION THREAD

**Discussion by oddstr13** [jellyfin/jellyfin#4650] [General]
I personally think that `PascalCase` doesn't look good in the API, and would strongly prefer `snake_case` instead (no SCREAMING_SNAKE_CASE please).

It just feels weird to have (potentially) case sensitive parts of an URL (considering the host part is case insensitive), and it feels out of place in JSON keys too.

**Comment by nielsvanvelzen** [comment]
I think it makes sense to use PascalCase because most of the API currently is. Altough I would prefer snake_case. This should definitely be considered for the API rewrite.

**Reply by oddstr13** [reply]
This is intended as a discussion about what we *want* the API to look like for v11+ (and also for *new* API endpoints before that time, so we don't have to rewrite them).

**Comment by crobibero** [comment]
Routes, QueryParameters, and Json keys should already be case insentive. `snake_case` isn't currently supported by the Json library, and is not the standard for C# 

(I only speak for 10.7+)

**Reply by oddstr13** [reply]
I don't think what's standard or convention in C# itself matters much for how the APIs should look externally, but no support for it in the JSON library would be a concern if it isn't relatively straight forward to implement it.
